tests/tests.py

```diff
@@ -55,6 +55,10 @@ def test_multi_super(self):
             result, "block2 from test6 - block2 from test3 - block2 from test1"
         )
 
+    def test_multi_level_inherited_template_block(self):
+        result = render_block_to_string("test8_django.html", "block1")
+        self.assertEqual(result, "block3 from test8")
```

Comment:
I think the additional templates are unnecessary and this could be simplified to just:

```suggestion
    def test_multi_level_inherited_template_block(self):
        result = render_block_to_string("test4.html", "block1")
        self.assertEqual(result, "block1 from test1")
```

Does that fully exercise what you're attempting to fix here?